src/axom/primal/geometry/NURBSCurve.hpp

```diff
@@ -0,0 +1,1177 @@
+    return span + r;
+  }
+
+  /*!
+   * \brief Splits a NURBS curve into two curves at a given parameter value
+   *
+   * \param [in] t parameter value between 0 and 1 at which to evaluate
+   * \param [out] n1 First output NURBS curve
+   * \param [out] n2 Second output NURBS curve
+   * \param [in] normalize Whether to normalize the output curves
+   * 
+   * \pre Parameter \a t must be in the span of the knots
+   */
+  void split(T t,
+             NURBSCurve<T, NDIMS>& n1,
+             NURBSCurve<T, NDIMS>& n2,
+             bool normalize = false) const
+  {
+    SLIC_ASSERT(t >= m_knotvec[0] && t <= m_knotvec[m_knotvec.getNumKnots() - 1]);
+
+    const bool isCurveRational = this->isRational();
+    const int p = getDegree();
+
+    // Handle the special case of splgitting at the endpoints
+    if(t == 0.0)
+    {
+      n1.setParameters(p + 1, p);
+      for(int i = 0; i <= p; ++i)
+      {
+        n1[i] = m_controlPoints[0];
+        if(isCurveRational)
+        {
+          n1.makeRational();
+          n1.setWeight(i, m_weights[0]);
+        }
+      }
+      n2 = *this;
+      return;
+    }
+    else if(t == 1.0)
+    {
+      n1 = *this;
+      n2.clear();
+      n2.setParameters(p + 1, p);
+      for(int i = 0; i <= p; ++i)
+      {
+        n2[i] = m_controlPoints[getNumControlPoints() - 1];
+        if(isCurveRational)
+        {
+          n2.makeRational();
+          n2.setWeight(i, m_weights[getNumControlPoints() - 1]);
+        }
+      }
+      return;
+    }
+
+    n1 = *this;
+
+    // Will make the multiplicity of the knot equal to p,
+    //  even if it is already >= 1
+    auto s = n1.insertKnot(t, p);
+    int nkts = n1.getNumKnots();
+
+    // Split the knot vector
+    KnotVectorType k1, k2;
+    m_knotvec.split(s, k1, k2);
+
+    n1.m_knotvec = k1;
+    n2.m_knotvec = k2;
+
+    // Copy the control points
+    n2.m_controlPoints.resize(nkts - s - 1);
+    n2.m_weights.resize(isCurveRational ? nkts - s - 1 : 0);
+    for(int i = 0; i < n2.m_controlPoints.size(); ++i)
+    {
+      n2.m_controlPoints[nkts - s - 2 - i] =
+        n1.m_controlPoints[n1.m_controlPoints.size() - 1 - i];
+
+      if(isCurveRational)
+      {
+        n2.m_weights[nkts - s - 2 - i] =
+          n1.m_weights[n1.m_weights.size() - 1 - i];
+      }
+    }
+
+    n1.m_controlPoints.resize(s - p + 1);
+    n1.m_weights.resize(isCurveRational ? s - p + 1 : 0);
+
+    if(normalize)
+    {
+      n1.normalize();
+      n2.normalize();
+    }
+  }
+
+  /// \brief Normalize the parameterization of the knot vector
+  void normalize() { m_knotvec.normalize(); }
+
+  /*!
+   * \brief Splits a NURBS curve (at each internal knot) into several Bezier curves
+   *   
+   * \return An array of Bezier curves
+   */
+  axom::Array<BezierCurve<T, NDIMS>> extractBezier() const
+  {
+    axom::Array<BezierCurve<T, NDIMS>> beziers;
+
+    const bool isCurveRational = this->isRational();
+
+    int p = getDegree();
+    if(p == 0)  // Handle this special case
```

Comment:
Is `p==0` a collection of disconnected points? Please add a comment.

Reply:
Yes, that's right. The degree 0 case is the only exception to the "curves must be continuous" rule, and is really only permitted because we allow order 0 Bezier curves that only have a single control point. I've added this exception to the `\brief` for `NURBSCurve`.